telescope/filters.py

```diff
@@ -82,212 +91,214 @@ def _filter_c2s_measurement(measurement):
 
+def _filter_s2c_measurement(measurement):
+  """Validates result of a server-to-client test.
+
+  Applies measurement validity rules and tests presence of required fields for
+  a download (server-to-client) test.
+
+  Args:
+    measurement: (dict) Measurement Lab and web100 variables for test.
+
+  Returns:
+    bool: True if valid measurement, False otherwise.
+
+  Required Measurements Fields:
+    * web100_log_entry.snap.SndLimTimeRwin
+    * web100_log_entry.snap.SndLimTimeCwnd
+    * web100_log_entry.snap.SndLimTimeSnd
+    * web100_log_entry.snap.CongSignals
+    * web100_log_entry.snap.HCThruOctetsAcked
+    * connection_spec.data_direction
+  Validity Rules:
+    Must last for more than, or equal to, 9 seconds and be less than, or
+      equal to one hour.
+        * (SndLimTimeRwin + SndLimTimeCwnd + SndLimTimeSnd) >= 9000000
+        * (SndLimTimeRwin + SndLimTimeCwnd + SndLimTimeSnd) < 3600000000
+    Must exchange at least 8192 bytes.
+        * HCThruOctetsAcked >= 8192
+    Must leave slow start phase of TCP, through reaching Congestion at
+      least once.
+        * CongSignals > 0
   """
+  required_fields = ['web100_log_entry_snap_SndLimTimeRwin',
+                     'web100_log_entry_snap_SndLimTimeCwnd',
+                     'web100_log_entry_snap_SndLimTimeSnd',
+                     'web100_log_entry_snap_CongSignals',
+                     'web100_log_entry_snap_HCThruOctetsAcked',
+                     'connection_spec_data_direction']
 
-  required_fields = ['web100_log_entry_snap_SndLimTimeRwin', 'web100_log_entry_snap_SndLimTimeCwnd',
-                      'web100_log_entry_snap_SndLimTimeSnd', 'web100_log_entry_snap_CongSignals',
-                      'web100_log_entry_snap_HCThruOctetsAcked', 'connection_spec_data_direction']
-
-  if False in [required_field in measurement.keys() for required_field in required_fields]:
+  if False in [required_field in measurement for required_field in required_fields]:
     raise ValueError('MissingField')
 
   assert int(measurement['connection_spec_data_direction']) == 1
 
   if not ((int(measurement['web100_log_entry_snap_SndLimTimeRwin']) +
-                              int(measurement['web100_log_entry_snap_SndLimTimeCwnd']) +
-                              int(measurement['web100_log_entry_snap_SndLimTimeSnd'])) >= _MINIMUM_DURATION):
+           int(measurement['web100_log_entry_snap_SndLimTimeCwnd']) +
+           int(measurement['web100_log_entry_snap_SndLimTimeSnd']))
+          >= _MINIMUM_DURATION):
     return False
   if not ((int(measurement['web100_log_entry_snap_SndLimTimeRwin']) +
-                            int(measurement['web100_log_entry_snap_SndLimTimeCwnd']) +
-                            int(measurement['web100_log_entry_snap_SndLimTimeSnd'])) < _MAXIMUM_DURATION):
+           int(measurement['web100_log_entry_snap_SndLimTimeCwnd']) +
+           int(measurement['web100_log_entry_snap_SndLimTimeSnd']))
+          < _MAXIMUM_DURATION):
```

Comment:
total_time = (int(measurement['web100_log_entry_snap_SndLimTimeRwin']) +  int(measurement['web100_log_entry_snap_SndLimTimeCwnd']) + int(measurement['web100_log_entry_snap_SndLimTimeSnd']))

if not (_MINIMUM_DURATION <= total_time < _MAXIMUM_DURATION):
  return False